### ISSUE TITLE
Unexported capitalized constant is considered ALL_CAPS instead of CamelCase

### ISSUE THREAD

**Issue by ChimeraCoder** [golang/lint#32]
The `_POST` declaration in the following snippet  gives me the error `twitter.go:55:2: don't use ALL_CAPS in Go names; use CamelCase`

``` go
const (
        _GET  = iota
        _POST = iota
)
```

Full code: https://github.com/ChimeraCoder/anaconda/blob/master/twitter.go#L53-L57

(Interestingly, `_GET` does not return an error, though it does if I add any characters to the name - I guess it is special-cased somehow).

I feel that this style (a capitalized constant, prepended with an underscore to prevent it from being exported) should not be considered `ALL_CAPS` for the purposes of golint - I'm not sure what the alternative CamelCase would be, since this is a single word

I'm filing an issue rather than submitting a pull request in case there is a reason for this style suggestion that I'm not aware of, but as it stands, this error message seems either incorrect or misleadingly named.
